### ISSUE TITLE
please make hash algorithm configurable

### ISSUE THREAD

**Issue by abraxxa** [jonathanstowe/RedX-HashedPassword#5]
Hi,
please make the hash algorithm configurable so one can be sure that no lesser secure hashes sneak into the database.
Having a way to upgrading hashes when this setting changes would also be welcome.
This should still be able to check a hash stored with a previous algorithm but update it to the current on check-password.

Thanks!

**Comment by jonathanstowe** [comment]
> Having a way to upgrading hashes when this setting changes would also be welcome

I'm afraid that wouldn't be possible as it would require the original plaintext password, the only way to "upgrade" would be to update with a new password.

**Comment by abraxxa** [comment]
Yes, when check-password is called you habe the cleartext password to do that.
